superduperdb/base/document.py

```diff
@@ -145,6 +144,12 @@ def decode(
             db=db,
         )
 
+        # Then decode the data with the schema
+        if isinstance(r, dict):
+            if schema is not None:
+                schema.init()
+                r = schema.decode_data(r)
+
```

Comment:
To fix the order of decoding, we should first restore the leaf and then decode using the schema.

Because during encoding, we first encode using the schema and then encode the leaf.

Reply:
I don't follow.

Reply:
Encoding Process

Data -> schema.encode -> leaf.encode -> Encoded Data

Old Decoding Process

Encoded Data -> schema.decode -> leaf.decode

New Decoding Process

Encoded Data -> leaf.decode -> schema.decode